libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaNativeLibraryProvider.java

```diff
@@ -8,13 +8,26 @@
 
 package org.elasticsearch.nativeaccess.jna;
 
+import org.elasticsearch.nativeaccess.lib.JavaLibrary;
 import org.elasticsearch.nativeaccess.lib.NativeLibraryProvider;
 import org.elasticsearch.nativeaccess.lib.PosixCLibrary;
+import org.elasticsearch.nativeaccess.lib.ZstdLibrary;
 
 import java.util.Map;
 
 public class JnaNativeLibraryProvider extends NativeLibraryProvider {
+
     public JnaNativeLibraryProvider() {
-        super("jna", Map.of(PosixCLibrary.class, JnaPosixCLibrary::new));
+        super(
+            "jna",
+            Map.of(
```

Comment:
Nit: if we think we will continue to add libraries, maybe Map.ofEntries could produce a better formatting here.

Reply:
I'll consider changing that in a followup, you are right it might look nicer, and there are several more of these to come.